client/extensions/woocommerce/state/sites/products/reducer.js

```diff
@@ -34,5 +32,6 @@ function createProductSuccess( state, action ) {
 		newProducts.push( product );
 	}
 
-	return state;
+	return { ...state, product: newProducts };
```

Comment:
@coderkevin Sorry for not catching this before, but this should be `products: newProducts`.